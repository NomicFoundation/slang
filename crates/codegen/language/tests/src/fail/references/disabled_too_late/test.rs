#![allow(unused_crate_dependencies)]

codegen_language_macros::compile!(Language(
    name = Foo,
    root_item = One,
    leading_trivia = Sequence([]),
    trailing_trivia = Sequence([]),
    versions = ["1.0.0", "2.0.0", "3.0.0", "4.0.0", "5.0.0"],
    sections = [Section(
        title = "Section One",
        topics = [Topic(
            title = "Topic One",
            items = [
                Struct(
                    name = One,
                    fields = (
                        field_1 = Optional(
                            kind = NonTerminal(Two),
                            enabled_in = "2.0.0",
                            disabled_in = "3.0.0"
                        ),
                        field_2 = Optional(kind = Terminal([Three]))
                    )
                ),
                Struct(
                    name = Two,
                    enabled_in = "2.0.0",
                    disabled_in = "3.0.0",
                    fields = (field_1 = Optional(kind = Terminal([Three]), disabled_in = "4.0.0"))
                ),
                Token(
                    name = Three,
                    definitions = [TokenDefinition(scanner = Atom("three"))]
                )
            ]
        )]
    )]
));

fn main() {}
