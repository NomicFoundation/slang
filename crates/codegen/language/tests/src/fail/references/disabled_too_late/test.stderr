error: Parent scope is only enabled in '2.0.0..3.0.0'.
  --> src/fail/references/disabled_too_late/test.rs:28:81
   |
28 |                         (field_1 = Optional(kind = Terminal([Three]), enabled = Till("4.0.0")))
   |                                                                                 ^^^^
